### ISSUE TITLE
Arcane Focus mage talent weird behaviour

### ISSUE THREAD

**Issue by haebawari** [chromiecraft/chromiecraft#3497]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

If I understand the hit rules correctly, with 7%+ spell hit you should never see misses on your spells in PVP. However, having 4.6% base hit and 3 additional percent from the arcane hit talent I still see misses on Arcane Barrage pretty regularly. When I had base 5% hit or more without the talent, I'm pretty sure I haven't seen a single miss.

### Expected Blizzlike Behaviour

"Arcane Focus" hit rating stacking with your base hit rating (for arcane spells)

### Source

http://forum.warmane.com/showthread.php?t=334055

### Steps to reproduce the problem

1. Have a mage with 3% - 4.6% base hit rating
2. Put 3 talent points into Arcane Focus
3. Observe misses with arcane spells in PVP
4.  Have flat 6% base hit rating or more
5. No PVP misses with arcane spells

### Extra Notes

My data might be biased or scuffed, additionally my understanding of the hit behaviour might be wrong. If so, please clarify.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/c7f74e563758241e29721e95b86aed0093a8856a

### Operating system

win 7

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by ishidinho** [comment]
Hey @haebawari, thank you for the report. 

I closed a similar ticket https://github.com/chromiecraft/chromiecraft/issues/3042 some months ago, with the argument of [Spell Hit and Resistance](https://wowwiki-archive.fandom.com/wiki/Spell_hit).

TL;DR - there are two rolls for spell hit. First, spell hit itself. Second, spell resistance. So, if your target has enough resistance, there is a further chance to miss the target, even of your are massively overcapped. 

I close this ticket. If you have clear evidence that something is wrong with spell hit an resistance, feel free to open a ticket again.

**Comment by haebawari** [comment]
So what you're saying is that I should still see misses with arcane spells on blood elf retri paladins and such even with 10.6% hit? (7.6% from items + 3% from Arcane Focus)? Because if that's what you're saying, then my data so far does not support this. My evidence was that when I gear for 6%+ base spell hit I see no misses. On other hand, if I have 4.6% base hit + 3% from talents I see misses even on arcane spells on average 1-2 times per bg.
I'll keep updating if I find a counterargument tho.

**Comment by ishidinho** [comment]
> Also, you mentioned spell resistance as the second roll. Wouldn't this effect if happening result in a "resist" pop up instead of a "miss" one?

Good point. My bad. I reopen the report for further investigations.

**Comment by Gultask** [comment]
I think I saw somewhere about resists sometimes showing up as misses in another pserver so I'll investigate it as well.

Meanwhile, I tried out the Arcane Focus spell and it seems to be working fine.
I've cast 800 arcane spells while having 4.25% hit + Arcane Focus 3/3 and saw 0 misses.

If I find more info I'll post it here.

**Comment by haebawari** [comment]
Been trying to reproduce on Alterac Valley (the only bg happening rn) for 2 days with no avail. The only misses were on non-arcane spells vs rogues and blood elf paladins, which is to be expected with my base hit.
I really don't think I was imagining those arcane misses tho, they happened on Arcane Barrage, Arcane Missiles and Slow as well with the same gear and talents. I saw the misses in recount but I didn't screenshot them. That was on Warsong Gulch, and one of the misses happened on druid in bear form, who doesn't have any +miss talents abilities afaik.

**Comment by haebawari** [comment]
![awdada](https://user-images.githubusercontent.com/61579401/169715559-e90e1fb8-7cf3-406b-989b-624f28886283.png)
So here's some misses I spotted. These four happened on paladins and rogues, who might have additionally been blood elves. I'm aware that paladins and rogues have talents "Divine Purpose" and "Heightened Senses" which make them 4% less likely to get hit by spells. My hit rating is 4.6 + 3 from the talent. 
In this case is it true that my hit rating isn't enough to completely avoid misses? Chance to miss is 4% for a player your level + 2 % for Blood Elf + 4% for paladins and rogues = 10%? Also would these misses show up as misses or resists?

**Comment by Gultask** [comment]
>You need 165 hit rating to reach 5% hit to negate base miss rate in pvp. You'd need an additional 5% to negate talents like [arctic winds](https://www.wowhead.com/?spell=31678)/[heightened senses](https://www.wowhead.com/?spell=30895), however I'm not sure if 300+ hit rating is worth striving for in pvp gear. ([source](https://www.wowhead.com/forums/topic/pvp-hit-and-miss-118412#p1370480))

>For non-binary spells, resistance is computed as a second calculation after the spell hit chance is rolled. After a spell hit is determined, a partial resist calculation based on resistance is done, which can also result in a full resist. These two resists once appeared identically in the combat log, and there was no known way to distinguish a spell miss from a full resist. However, as of [patch 3.0.2](https://wowwiki-archive.fandom.com/wiki/Patch_3.0.2) **the combat log now distinguishes between a spell miss and a resist.** ([source](https://wowwiki-archive.fandom.com/wiki/Spell_hit?oldid=2414471))

>However, there are some border cases; for example, the Hunter's  [[Arcane Shot]](https://wowpedia.fandom.com/wiki/Arcane_Shot) can be partially resisted, but a full resist is shown as a miss in the combat log. ([source](https://wowpedia.fandom.com/wiki/Resist?oldid=2217960))

Yes, I'm pretty sure the miss chance would be calculated like that, giving you a chance to miss. And those would be misses and not resists.
This is why I don't like PvP, there's so many little things that I didn't even know existed 😄 
So I guess it's working correctly after all. Thanks for the feedback, haebawari!
